engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -1235,4 +1235,30 @@ function updateLastAction($guid, $posted = null) {
 			return false;
 		}
 	}
+
+	/**
+	 * Get a user by GUID even if the entity is hidden or disabled
+	 *
+	 * @param int $guid User GUID. Default is logged in user
+	 *
+	 * @return \ElggUser|false
+	 * @access private
+	 */
+	public function resolveUser($guid = 0) {
```

Comment:
Can we make this more generic?
I think it would be useful to have something like `resolve($guid = 0, $type = null, $subtype = null)`. We could then use `elgg_entity_exists()` beforehand, or `elgg_instanceof()` afterwards.